File: infinite-herbs-backend/app/models/role.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.database import Base
from .role_permissions import role_permissions

class Role(Base):
    __tablename__ = "roles"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String, unique=True, nullable=False, index=True)
    description = Column(String, nullable=True)
    createdAt = Column(DateTime(timezone=True), server_default=func.now())
    updatedAt = Column(DateTime(timezone=True), onupdate=func.now())

    users = relationship("User", back_populates="role")

    permissions = relationship(
        "Permission",
        secondary=role_permissions,
        back_populates="roles",
    )

File: infinite-herbs-backend-.net/Models/Role.cs
using System.ComponentModel.DataAnnotations;
namespace AdminApi.Models;
public class Role {
    public int Id { get; set; }
    [Required] public string Name { get; set; } = "";
    public string Description { get; set; } = "";
    public ICollection<RolePermission> Permissions { get; set; } = new List<RolePermission>();
    public ICollection<User> Users { get; set; } = new List<User>();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}